xrpl/core/binarycodec/types/number.py

```
@@ -42,17 +55,49 @@ def normalize(mantissa: int, exponent: int) -> Tuple[int, int]:
+
+    if m > _MAX_REP:
+        if _DEBUG:
+            print(
+                "Mantissa "
+                + str(m)
+                + " is larger than _MAX_REP. Reducing the mantissa while increasing the exponent...\n"
+            )
+        if exponent > _MAX_EXPONENT:
+            raise XRPLBinaryCodecException(
+                "Overflow: Mantissa and exponent are too large."
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)